scaletest/workspacetraffic/run_test.go

```diff
@@ -226,12 +238,32 @@ func TestRun(t *testing.T) {
 		})
 
 		var logs strings.Builder
-		// Stop the test after one 'tick'. This will cause an EOF.
+
+		runDone := make(chan struct{})
 		go func() {
-			<-time.After(cancelAfter)
-			cancel()
+			defer close(runDone)
+			err := runner.Run(ctx, "", &logs)
+			assert.NoError(t, err, "unexpected error calling Run()")
 		}()
-		require.NoError(t, runner.Run(ctx, "", &logs), "unexpected error calling Run()")
+
+		gotMetrics := make(chan struct{})
+		go func() {
```

Comment:
nit: I'm always confused if there is a possibility of a routine leak. defer goes first then routine is killed, right?

Reply:
Defer runs when the surrounding function returns. So `assert.Eventually()` has to finish before `gotMetrics` is closed, and `<-gotMetrics` will block until that happens.